src/client/cmd/common_cli.cpp

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright (C) 2018 Canonical, Ltd.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; version 3.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "common_cli.h"
+#include "exec.h"
+
+#include <multipass/cli/argparser.h>
+#include <multipass/cli/format_utils.h>
+
+#include <fmt/ostream.h>
+
+namespace mp = multipass;
+namespace cmd = multipass::cmd;
+
+namespace
+{
+mp::ReturnCode return_code_for(const grpc::StatusCode& code)
+{
+    return code == grpc::StatusCode::UNAVAILABLE ? mp::ReturnCode::DaemonFail : mp::ReturnCode::CommandFail;
+}
+} // namespace
+
+mp::ParseCode cmd::handle_all_option(ArgParser* parser, std::ostream& cerr)
```

Comment:
This handles either instance name or --all being used, function name not great...

Reply:
Hmm, yeah, so maybe something like `check_for_name_and_all_option_conflict()`?  Seems kind of verbose, but that is really what it is doing:)

Reply:
Yep, is verbose but not that bad. I'd vote for it